presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixClient.java

```diff
@@ -0,0 +1,453 @@
+    {
+        String arrayTypeName = arrayTypeHandle.getJdbcTypeName()
+                .orElseThrow(() -> new PrestoException(JDBC_ERROR, "Type name is missing: " + arrayTypeHandle));
+        checkArgument(arrayTypeName.endsWith(" ARRAY"), "array type must end with ' ARRAY'");
+        arrayTypeName = arrayTypeName.substring(0, arrayTypeName.lastIndexOf(" ARRAY"));
+        return new JdbcTypeHandle(
+                PDataType.fromSqlTypeName(arrayTypeName).getSqlType(),
+                Optional.of(arrayTypeName),
+                arrayTypeHandle.getColumnSize(),
+                arrayTypeHandle.getDecimalDigits(),
+                arrayTypeHandle.getArrayDimensions());
+    }
+
+    public List<JdbcColumnHandle> getNonRowkeyColumns(ConnectorSession session, JdbcTableHandle tableHandle)
+    {
+        return getColumns(session, tableHandle).stream().filter(handle -> !PhoenixMetadata.ROWKEY.equalsIgnoreCase(handle.getColumnName())).collect(Collectors.toList());
+    }
+
+    public Optional<String> getCatalogName()
+    {
+        // catalogName in Phoenix is used for tenantId, but
+        // tenant-specific connections not currently supported)
+        return Optional.of("");
+    }
+
+    public QueryPlan getQueryPlan(PhoenixPreparedStatement inputQuery)
+    {
+        try {
+            // Optimize the query plan so that we potentially use secondary indexes
+            QueryPlan queryPlan = inputQuery.optimizeQuery();
+            // Initialize the query plan so it sets up the parallel scans
+            queryPlan.iterator(MapReduceParallelScanGrouper.getInstance());
+            return queryPlan;
+        }
+        catch (SQLException e) {
+            throw new PrestoException(PHOENIX_QUERY_ERROR, "Failed to get the Phoenix query plan", e);
+        }
+    }
+
+    private ResultSet getResultSet(InputSplit split, QueryPlan queryPlan)
+    {
+        List<Scan> scans = ((PhoenixInputSplit) split).getScans();
+        try {
+            List<PeekingResultIterator> iterators = new ArrayList<>(scans.size());
+            StatementContext context = queryPlan.getContext();
+            // Clear the table region boundary cache to make sure long running jobs stay up to date
+            byte[] tableNameBytes = queryPlan.getTableRef().getTable().getPhysicalName().getBytes();
+            ConnectionQueryServices services = context.getConnection().getQueryServices();
+            services.clearTableRegionCache(tableNameBytes);
+
+            long renewScannerLeaseThreshold = context.getConnection().getQueryServices().getRenewLeaseThresholdMilliSeconds();
+            for (Scan scan : scans) {
+                // For MR, skip the region boundary check exception if we encounter a split. ref: PHOENIX-2599
+                scan.setAttribute(SKIP_REGION_BOUNDARY_CHECK, Bytes.toBytes(true));
+
+                PeekingResultIterator peekingResultIterator;
```

Comment:
Move the variable declaration below to where it is initialized, or simply inline the expression (it is short)